plugins/modules/git_config_info.py

```diff
@@ -0,0 +1,183 @@
+    scope: file
+    path: /etc/gitconfig
+'''
+
+RETURN = '''
+---
+config_value:
+  description: >
+    When O(name) is set, a string containing the value of the setting in name. If O(name) is not set, empty.
+    If a conifg key (e.g. push.pushoption) has more then one entry, just the first one is returned here.
+  returned: success
+  type: str
+  sample: "vim"
+
+config_values:
+  description: >
+    When O(name) is not set, a dict containing key/List[value] pairs of all configuration settings.
+    When O(name) is set, a dict containing key/List[value] pairs of the for the defined key.
+  returned: success
+  type: dict
+  sample:
+    core.editor: ["vim"]
+    color.ui: ["auto"]
+    push.pushoption: ["merge_request.create", "merge_request.draft"]
+    alias.remotev: ["remote -v"]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type="str"),
+            path=dict(type="path"),
+            scope=dict(required=False, type="str", default="system", choices=["global", "system", "local", "file"]),
+        ),
+        required_if=[
+            ("scope", "local", ["path"]),
+            ("scope", "file", ["path"])
+        ],
+        required_one_of=[],
+        supports_check_mode=True,
+    )
+
+    # We check error message for a pattern, so we need to make sure the messages appear in the form we're expecting.
+    # Set the locale to C to ensure consistent messages.
+    module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
+
+    name = module.params["name"]
+    path = module.params["path"]
+    scope = module.params["scope"]
+
+    run_cwd = path if scope == "local" else "/"
+    args = build_args(module, name, path, scope)
+
+    (rc, out, err) = module.run_command(args, cwd=run_cwd, expand_user_and_vars=False)
+
+    if rc == 128 and "unable to read config file" in err:
+        # This just means nothing has been set at the given scope
+        pass
+    elif rc >= 2:
+        # If the return code is 1, it just means the option hasn't been set yet, which is fine.
+        module.fail_json(rc=rc, msg=err, cmd=" ".join(args))
+
+    output_lines = out.rstrip().splitlines()
+
+    if name:
+        first_value = output_lines[0] if output_lines else ""
+        config_values = {name: output_lines}
+        module.exit_json(changed=False, msg="", config_value=first_value, config_values=config_values)
+    else:
+        config_values = text_to_dict(output_lines)
+        module.exit_json(changed=False, msg="", config_value="", config_values=config_values)
+
+
+def build_args(module, name, path, scope):
+    git_path = module.get_bin_path("git", True)
+    args = [git_path, "config", "--includes", "--" + scope]
+
+    if scope == "file":
+        args.append(path)
+
+    if name:
+        args.extend(["--get-all", name])
+    else:
+        args.append("--list")
+
```

Comment:
Why not always append `--null` as well? This makes parsing the result a lot safer (you need to split by `\0` though).

Reply:
This is a great idea